bot/core/converters.py

```diff
@@ -63,41 +81,132 @@ async def convert(self, ctx: Context, message: str) -> str:
-    When possible try to convert user into `Member` but if not, use `User` instead.
+        duration_dict = {unit: int(amount) for unit, amount in time_delta.groupdict(default=0).items()}
+        return relativedelta(**duration_dict)
+
+
+class CodeBlock(Converter):
+    """
+    Convert given wrapped string in codeblock into a tuple of language and the wrapped string
     """
 
-    async def get_member(guild: Guild, user: User) -> Member:
-        member = guild.get_member(user.id)
-        if not member:
-            try:
-                member = await guild.fetch_member(user.id)
-            except NotFound:
-                raise MemberNotFound(f"No member with ID: {user.id} on guild {guild.id}")
-        return member
+    codeblock_parser = re.compile(r"\`\`\`(.*\n)?((?:[^\`]*\n*)+)\`\`\`")
+    inline_code_parser = re.compile(r"\`(.*\n*)\`")
 
-    async def convert(self, ctx: Context, argument: str) -> Member:
-        """Convert the `argument` into `Member` or `User`."""
+    def convert(self, ctx: Context, codeblock: str) -> t.Tuple[t.Optional[str], str]:
+        """
+        Convert a string `codeblock` into a tuple which consists of:
+        * language (f.e.: `py` or `None` for no language)
+        * wrapped_text (the text inside of the codeblock)
+
+        The converter converts:
+        * full codeblocks (```text```, ```py\ntext```, ```\ntext```)
+        * inline codeblocks (`text`)
+
+        In case no codeblock was found, original string is returned
+        """
+        codeblock_match = self.codeblock_parser.fullmatch(codeblock)
+        if codeblock_match:
+            lang = codeblock_match.group(1)
+            code = codeblock_match.group(2)
+            if not code:
+                code = lang
+                lang = None
+            if code[-1] == "\n":
+                code = code[:-1]
+            return (lang, code)
+
+        inline_match = self.inline_code_parser.fullmatch(codeblock)
+        if inline_match:
+            return (None, inline_match.group(1))
+
+        return codeblock
+
+
+class ProcessedUser(UserConverter):
+    """
+    Try to convert any accepted string into `User`
+    Lookup Strategy:
+    [Default UserConverter strategy]
+    1. Lookup by ID.
+    2. Lookup by mention.
+    3. Lookup by name#discrim
+    4. Lookup by name
+    [Added functionality]
+    5. Lookup by API
+    """
+
+    async def convert(self, ctx: Context, argument: str) -> User:
+        # Follow general MemberConverter lookup strategy
         with suppress(BadArgument):
-            # Try to use UserConverter first
-            user = await super().convert(ctx, argument)
-            try:
-                return await self.get_member(ctx.guild, user)
-            except MemberNotFound:
-                return user
+            return await super().convert(ctx, argument)
 
-        # If UserConverter failed, try to fetch user as ID
+        # Try to look user up using API
+        ID = _obtain_user_id(argument)
```

Comment:
where is this defined? same goes for line `206`

Reply:
On the top of the file?
https://github.com/The-Codin-Hole/HotWired-Bot/blob/3dc68ea845b3e0aba889ea60857aec09237cff5e/bot/core/converters.py#L16-L26